pkg/tests/tasks/security/certificate/cert_manager_test.go

```diff
@@ -6,68 +6,98 @@ import (
 
 	"github.com/maistra/maistra-test-tool/pkg/app"
 	"github.com/maistra/maistra-test-tool/pkg/util/check/assert"
+	"github.com/maistra/maistra-test-tool/pkg/util/env"
 	"github.com/maistra/maistra-test-tool/pkg/util/helm"
+	"github.com/maistra/maistra-test-tool/pkg/util/ns"
 	"github.com/maistra/maistra-test-tool/pkg/util/oc"
 	"github.com/maistra/maistra-test-tool/pkg/util/pod"
 	"github.com/maistra/maistra-test-tool/pkg/util/retry"
 	"github.com/maistra/maistra-test-tool/pkg/util/test"
+	"github.com/maistra/maistra-test-tool/pkg/util/version"
 )
 
 func TestCertManager(t *testing.T) {
 	test.NewTest(t).Id("T38").Groups(test.Full, test.ARM, test.InterOp).Run(func(t test.TestHelper) {
-		ns := "foo"
-		certManagerNs := "cert-manager"
+		smcpVer := env.GetSMCPVersion()
+		if smcpVer.LessThan(version.SMCP_2_3) {
+			t.Skip("cert-manager-istio-csr is not supported in SMCP older than v2.3.2")
+		}
 
+		certManagerOperatorNs := "cert-manager-operator"
+		certManagerNs := "cert-manager"
 		t.Cleanup(func() {
 			helm.Namespace(meshNamespace).Release("istio-csr").Uninstall(t)
-			helm.Namespace(certManagerNs).Release("cert-manager").Uninstall(t)
+			oc.DeleteFromString(t, meshNamespace, istioCA)
+			oc.DeleteFromString(t, certManagerNs, rootCA)
+			oc.DeleteFromString(t, certManagerOperatorNs, certManagerOperator)
+			oc.DeleteSecret(t, meshNamespace, "istiod-tls")
+			oc.DeleteSecret(t, meshNamespace, "istio-ca")
 			oc.DeleteNamespace(t, certManagerNs)
-			oc.RecreateNamespace(t, ns)
+			oc.DeleteNamespace(t, certManagerOperatorNs)
+			oc.RecreateNamespace(t, ns.Foo)
 		})
 
 		t.LogStep("Uninstall existing SMCP")
 		oc.RecreateNamespace(t, meshNamespace)
 
-		t.LogStep("Create namespace for cert-manager")
-		oc.CreateNamespace(t, certManagerNs)
-
-		t.LogStep("Add jetstack repo to helm")
-		helm.Repo("https://charts.jetstack.io").Add(t, "jetstack")
+		t.LogStep("Create namespace for cert-manager-operator")
+		oc.CreateNamespace(t, certManagerOperatorNs)
 
-		t.LogStep("Install cert-manager")
-		helm.Namespace(certManagerNs).
-			Chart("jetstack/cert-manager").
-			Release("cert-manager").
-			Version("v1.11.0").
-			Set("installCRDs=true").
-			Install(t)
-		oc.WaitPodsReady(t, certManagerNs, "app=cert-manager")
+		t.LogStep("Install cert-manager-operator")
+		oc.ApplyString(t, certManagerOperatorNs, certManagerOperator)
```

Comment:
Unfortunately, installation of cert-manager-operator is much slower than cert-manager from helm chart, but it's important from the product perspective to test integration with our operator rather than the community helm chart.